classes/plugins/PKPPubIdPlugin.inc.php

```diff
@@ -0,0 +1,410 @@
+	 * @return string
+	 */
+	abstract function getPubIdDisplayType();
+
+	/**
+	 * Full name of the public identifier.
+	 * @return string
+	 */
+	abstract function getPubIdFullName();
+
+	/**
+	 * Get the whole resolving URL.
+	 * @param $contextId integer
+	 * @param $pubId string
+	 * @return string resolving URL
+	 */
+	abstract function getResolvingURL($contextId, $pubId);
+
+	/**
+	 * Get the file (path + filename)
+	 * to be included into the object's
+	 * metadata pages.
+	 * @return string
+	 */
+	abstract function getPubIdMetadataFile();
+
+	/**
+	 * Get the class name of the settings form.
+	 * @return string
+	 */
+	abstract function getSettingsFormName();
+
+	/**
+	 * Get the additional form field names.
+	 * @return array
+	 */
+	abstract function getFormFieldNames();
+
+	/**
+	 * Get the the prefix form field name.
+	 * @return string
+	 */
+	abstract function getPrefixFieldName();
+
+	/**
+	 * Get the the suffix form field name.
+	 * @return string
+	 */
+	abstract function getSuffixFieldName();
+
+	/**
+	 * Get the the suffix patterns form field names.
+	 * @return array (pub object type => suffix pattern field name)
+	 */
+	abstract function getSuffixPatternsFieldNames();
+
+	/**
+	 * Get the checkbox form field name that
+	 * is used to define if a pub object should
+	 * be excluded from assigning the pub id to it.
+	 * @return string
+	 */
+	abstract function getExcludeFormFieldName();
+
+	/**
+	 * Get additional field names to be considered for storage.
+	 * @return array
+	 */
+	abstract function getDAOFieldNames();
+
+	/**
+	 * Get the possible publication object types.
+	 * @return array
+	 */
+	abstract function getPubObjectTypes();
+
+	/**
+	 * Get all publication objects of the given type.
+	 * @param $pubObjectType object
+	 * @param $contextId integer
+	 * @return array
+	 */
+	abstract function getPubObjects($pubObjectType, $contextId);
+
+	/**
+	 * Is this object type enabled in plugin settings
+	 * @param $pubObjectType object
+	 * @param $contextId integer
+	 * @return boolean
+	 */
+	abstract function isObjectTypeEnabled($pubObjectType, $contextId);
+
+	/**
+	 * Should the object be excluded from
+	 * assigning it the pub id
+	 * @param $pubObject object
+	 * @return boolean
+	 */
+	function isObjectExcluded($pubObject) {
+		$excludeFormFieldName = $this->getExcludeFormFieldName();
+		$excluded = $pubObject->getData($excludeFormFieldName);
+		return $excluded;
+	}
+
+	/**
+	 * Get the error message for not unique pub id
+	 * @return string
+	 */
+	abstract function getNotUniqueErrorMsg();
+
+	/**
+	 * Verify form data.
+	 * @param $fieldName string The form field to be checked.
+	 * @param $fieldValue string The value of the form field.
+	 * @param $pubObject object
+	 * @param $contextId integer
+	 * @param $errorMsg string Return validation error messages here.
+	 * @return boolean
+	 */
+	function verifyData($fieldName, $fieldValue, &$pubObject, $contextId, &$errorMsg) {
```

Comment:
Not sure the reference on pubObject is required (though I guess the one on errorMsg still is).